libfwupd/fwupd-device.c

```diff
@@ -169,9 +180,18 @@ fwupd_device_add_checksum(FwupdDevice *self, const gchar *checksum)
 
 	if (fwupd_device_has_checksum(self, checksum))
 		return;
+	fwupd_device_ensure_checksums(self);
```

Comment:
I think you want to push `fwupd_device_ensure_checksums()` _into_ `fwupd_device_has_checksum()`  Otherwise this can be a NULL pointer deref when it starts the loop.

Reply:
Nah, there's a:

	if (priv->checksums == NULL)
		return FALSE;

in fwupd_device_has_checksum(). I don't think it makes sense to create the array, to loop around the empty array, just to find out that there's nothing there. If there have never been any checksums added then fwupd_device_has_checksum() should always just return FALSE.